server-spi/src/main/java/org/keycloak/models/ClientScopeModel.java

```diff
@@ -93,5 +93,22 @@ default void setIncludeInTokenScope(boolean includeInTokenScope) {
         setAttribute(INCLUDE_IN_TOKEN_SCOPE, String.valueOf(includeInTokenScope));
     }
 
+    /**
+     * Indicates whether or not this scope is persistent. If {@code false} the scope is considered a transient scope, indicating that any reference
+     * to this scope should not be persisted (e.g: user consents)
+     *
+     * @return {@code true} if this is a persistent scope, otherwise {@code false}. Defaults to {@code true}
+     */
+    default boolean isPersistent() {
+        return true;
+    }
 
+    /**
```

Comment:
The ClientModel itself has a method "isDisplayOnConsentScreen()" . So you can probably just return false from this method in the openshift client and then the client itself won't be displayed on consent screen. Isn't it sufficient rather then adding this new property?

Reply:
I did that, please see latest changes. Was this review related with the `isPersistent` method ?

Reply:
It was related to method "isDefaultScope()" . I see it is used just in TokenManager. However is it any bad if you add client itself to clientScopes in TokenManager considering that it won't be shown on consent screen?

Reply:
Yes, which looks like a bug. I've created https://issues.jboss.org/browse/KEYCLOAK-8790 for it (more details in the email). I think that as easiest possibility, we can change the LoginActionsService.processConsent method to persist just those clientScopes, which require consent (clientScopeModel.isDisplayOnConsentScreen() )? This won't handle the corner case I described in the KEYCLOAK-8790 (Case when admin changed the clientScope to require consent in the meantime between consent screen was shown for the user and user confirmed it), however it is sufficient for most cases IMO.